src/validation.h

```diff
@@ -147,18 +147,29 @@ extern uint64_t nPruneTarget;
  * block is made active. Note that it does not, however, guarantee that the
  * specific block passed to it has been checked for validity!
  *
- * If you want to *possibly* get feedback on whether pblock is valid, you must
- * install a CValidationInterface (see validationinterface.h) - this will have
- * its BlockChecked method called whenever *any* block completes validation.
+ * Performs initial sanity checks using the provided BlockValidationState before
+ * connecting any block(s). If you want to *possibly* get feedback on whether
+ * pblock is valid beyond just cursory mutation/DoS checks, you must install
+ * a CValidationInterface (see validationinterface.h) - this will have its
+ * BlockChecked method called whenever *any* block completes validation (note
+ * that any invalidity returned via state will *not* also be provided via
+ * BlockChecked). There is, of course, no guarantee that any given block which
+ * is not a part of the eventual best chain will ever be checked.
```

Comment:
8fc192c

I think this whole comment can be made clearer. 

"You have 2 types of feedbacks:
- unconditional : submitted block validity with regards to mutation, anti-Dos is returned through `BlockValidationState`. It may be used for housekeeping as in `BlockProcessed`
- conditional : if caller subscribe to `CValidationInterface` and if submitted block passed those first checks, `ActivateBestChain`,  will be called to try to make progress towards the best known block and make it the tip of our block chain. Submitted block may be among such tip candidates. Consensus-validity feedback on best known block may be used for peer punishment and compact block promotion as in `BlockChecked`."